src/sonic-yang-mgmt/sonic_yang_ext.py

```diff
@@ -378,10 +378,17 @@ def _createLeafDict(self, model, table):
         #choices, this is tricky, since leafs are under cases in tree.
         choices = model.get('choice')
         if choices:
-            for choice in choices:
-                cases = choice['case']
+            # If single choice exists in container/list
+            if isinstance(choices, dict):
+                cases = choices['case']
```

Comment:
>case [](http://example.com/codeflow?start=33&length=4)

Is it possible that 'case' is not in choices dict? will it throw?

Reply:
My understanding that this should be validated by libyang.